graphql/resolvers.go

```diff
@@ -3930,48 +3930,39 @@ func (r *versionResolver) PreviousVersion(ctx context.Context, obj *restModel.AP
 }
 
 func (r *versionResolver) Status(ctx context.Context, obj *restModel.APIVersion) (string, error) {
-	failedAndAbortedStatuses := append(evergreen.TaskFailureStatuses, evergreen.TaskAborted)
-	opts := task.GetTasksByVersionOptions{
-		Statuses:                       failedAndAbortedStatuses,
-		FieldsToProject:                []string{task.DisplayStatusKey},
-		IncludeBaseTasks:               false,
-		IncludeBuildVariantDisplayName: false,
-	}
-	tasks, _, err := task.GetTasksByVersion(*obj.Id, opts)
-	if err != nil {
-		return "", InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch tasks for version: %s", err.Error()))
-	}
 	status, err := evergreen.VersionStatusToPatchStatus(*obj.Status)
 	if err != nil {
 		return "", InternalServerError.Send(ctx, fmt.Sprintf("An error occurred when converting a version status: %s", err.Error()))
 	}
+	isAborted := utility.FromBoolPtr(obj.Aborted)
+	nonAbortedStatuses := []string{}
+	if !isAborted {
+		nonAbortedStatuses = append(nonAbortedStatuses, status)
+	}
 	if evergreen.IsPatchRequester(*obj.Requester) {
 		p, err := data.FindPatchById(*obj.Id)
 		if err != nil {
 			return status, InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch Patch %s: %s", *obj.Id, err.Error()))
 		}
 		if len(p.ChildPatches) > 0 {
-			patchStatuses := []string{*p.Status}
 			for _, cp := range p.ChildPatches {
-				patchStatuses = append(patchStatuses, *cp.Status)
-				// add the child patch tasks to tasks so that we can consider their status
-				childPatchTasks, _, err := task.GetTasksByVersion(*cp.Id, opts)
+				cpVersion, err := model.VersionFindOneId(*cp.Version)
 				if err != nil {
-					return "", InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch tasks for patch: %s ", err.Error()))
+					return "", InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch version for patch: %s ", err.Error()))
+				}
+				if cpVersion.Aborted {
+					isAborted = true
+				} else {
+					nonAbortedStatuses = append(nonAbortedStatuses, *cp.Status)
 				}
-				tasks = append(tasks, childPatchTasks...)
 			}
-			status = patch.GetCollectiveStatus(patchStatuses)
+			status = patch.GetCollectiveStatus(nonAbortedStatuses)
 		}
 	}
 
-	taskStatuses := getAllTaskStatuses(tasks)
-
 	// If theres an aborted task we should set the patch status to aborted if there are no other failures
-	if utility.StringSliceContains(taskStatuses, evergreen.TaskAborted) {
-		if len(utility.StringSliceIntersection(taskStatuses, evergreen.TaskFailureStatuses)) == 0 {
-			status = evergreen.PatchAborted
-		}
+	if isAborted && len(utility.StringSliceIntersection(evergreen.TaskFailureStatuses, nonAbortedStatuses)) == 0 {
```

Comment:
Since child patch statuses are already in nonAbortedStatuses, aren't child patches already covered? What would a `ticket for downstream tasks to also show aborted instead of failure` cover?

Reply:
not too sure by what you mean by >>Since child patch statuses are already in nonAbortedStatuses, aren't child patches already covered?
Do you mean that we can just check if `status` is failing and not all nonabortedstatusus? because i think that does make sense

Reply:
Ah I understand now. Yes, I think it's a good idea to create a ticket and include the above photo. We should probably prioritize that ticket because I think this can end up confusing users. 
cc: @ablack12

Reply:
👍  assuming this is a UI ticket I think this will be up to them to prioritize (I'm not sure where that status comes from). Since this isn't yet a heavily used feature I don't know that it needs to be done ASAP but you're right that we shouldn't let it slip.

Reply:
I will include the UI changes regarding the aborted status badge in this ticket: https://jira.mongodb.org/browse/EVG-14586